src/goose.rs

```diff
@@ -1221,14 +1267,13 @@ impl GooseClient {
     ///         }
     ///     }
     /// ````
-    pub fn set_failure(&mut self, method: &GooseMethod, path: &str) {
-        let request_name = self.get_request_name(path);
-        let mut update_request = GooseRawRequest::new(method.clone(), &request_name);
-        update_request.success = false;
-        // This is an updaate to a previously recorded statistic.
-        update_request.update = true;
-
-        self.send_to_parent(update_request);
+    pub fn set_failure(&mut self, request: &mut GooseRawRequest) {
+        // Only send update if this was previously a success.
+        if request.success {
+            request.success = false;
+            request.update = true;
```

Comment:
Unless I miss something the update property gets never reset back - this points to a potential logic error — either it’s unnecessary or at least from set_success / set_failure it’s always true — which means it could save some memory by putting it into an argument and not persisting in the request struct.